nuxeo-retention-web/elements/nuxeo-retention-analytics.js

```diff
@@ -0,0 +1,365 @@
+            start-date="[[_formatDate(startDate)]]"
+            end-date="[[_extendEndDate(endDate)]]"
+            date-field="dc:created"
+            grouped-by="ecm:primaryType"
+            where='{"ecm:hasLegalHold": "true"}'
+            group-limit="20"
+            data="{{legalHoldByDocType}}"
+            index="[[index]]"
+          >
+          </nuxeo-repository-data>
+
+          <nuxeo-card heading="[[i18n('retentionAnalytics.legalHoldByDocType')]]">
+            <chart-pie
+              values="[[_values(legalHoldByDocType)]]"
+              labels="[[_labels(legalHoldByDocType)]]"
+              options="{_piechartOptions}"
+            >
+            </chart-pie>
+          </nuxeo-card>
+        </div>
+
+        <!-- Expiration next 12 months -->
+        <nuxeo-repository-data
+          start-date="[[_formatDateNow()]]"
+          end-date="[[_formatDateFutureInMonths(12)]]"
+          with-date-intervals="month"
+          date-field="ecm:retainUntil"
+          data="{{expirationPerMonth}}"
+          index="[[index]]"
+        >
+        </nuxeo-repository-data>
+
+        <nuxeo-card class="chartBar" heading="[[i18n('retentionAnalytics.expirationNext12Months.heading')]]">
+          <chart-bar
+            labels="[[_futureExpirationLabels(expirationPerMonth)]]"
+            values="[[_values(expirationPerMonth)]]"
+            options='{ "legend": { "display": false }, "animation": false }'
+          >
+          </chart-bar>
+        </nuxeo-card>
+      </nuxeo-page>
+    `;
+  }
+
+  static get is() {
+    return 'nuxeo-retention-analytics';
+  }
+
+  static get properties() {
+    return {
+      index: {
+        type: String,
+        value: '_all',
+      },
+      visible: Boolean,
+      startDate: {
+        type: String,
+        value: moment().subtract(1, 'year').format('YYYY-MM-DD'),
+      },
+      endDate: {
+        type: String,
+        value: moment().format('YYYY-MM-DD'),
+      },
+      _filterActive: {
+        type: Array,
+        value: [{ range: { 'ecm:retainUntil': { gte: moment().format('YYYY-MM-DD') } } }],
+      },
+      _filterExpired: {
+        type: Array,
+        value: [{ range: { 'record:retainUntil': { lte: moment().format('YYYY-MM-DD') } } }],
+      },
+      _filterInterminate: {
+        type: Array,
+        value: [{ range: { 'ecm:retainUntil': { gte: '9999-01-01T00:00:00.000Z' } } }],
+      },
+      _labelsRetentionStatus: {
+        type: Array,
+        value: ['No retention', 'Active retention', 'Expired retention', 'Interminate retention'],
+      },
+      _piechartOptions: {
+        type: Object,
+        value: { legend: { display: true, position: 'bottom', labels: { boxWidth: 12 } }, animation: false },
+      },
+    };
+  }
+
+  static get observers() {
+    return ['_resetStartDate(visible)'];
+  }
+
+  _resetStartDate(visible) {
+    // we trigger the elasticsearch by modifying the startDate each time the page is visible
```

Comment:
We don't have to specify that we use ES, perhpas it's not always the case. We have to be generic and don't add more info on the implementation where it's not needed.